vec/src/Data/Vec/DataFamily/SpineStrict.hs

```diff
@@ -845,6 +850,36 @@ ifoldr = getIFoldr $ N.induction1 start step where
 
 newtype IFoldr a n b = IFoldr { getIFoldr :: (Fin n -> a -> b -> b) -> b -> Vec n a -> b }
 
+scanr :: forall a b n. N.SNatI n => (a -> b -> b) -> b -> Vec n a -> Vec ('S n) b
+scanr f z = getScan $ N.induction1 start step where
+    start :: Scan a 'Z b
+    start = Scan $ \_ -> singleton z
+
+    step :: Scan a m b -> Scan a ('S m) b
+    step (Scan go) = Scan $ \(x ::: xs) -> let ys@(y ::: _) = go xs in f x y ::: ys
+
+newtype Scan a n b = Scan { getScan :: Vec n a -> Vec ('S n) b }
+
+scanl :: forall a b n. N.SNatI n => (b -> a -> b) -> b -> Vec n a -> Vec ('S n) b
+scanl f z = reverse . scanr (flip f) z . reverse
```

Comment:
This is not right. `scanl` is doable efficiently without a reverse.

Check e.g.

```haskell
scanl                   :: (b -> a -> b) -> b -> [a] -> [b]
scanl                   = scanlGo
  where
    scanlGo           :: (b -> a -> b) -> b -> [a] -> [b]
    scanlGo f q ls    = q : (case ls of
                               []   -> []
                               x:xs -> scanlGo f (f q x) xs)
```

Reply:
Oh, right. This also makes me realize that there's no reason to exclude `scanl` and `scanl1` from `Inline`, so I added them.